lib/auth/linkedin_oidc_authenticator.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+class Auth::LinkedInOidcAuthenticator < Auth::ManagedAuthenticator
+  class LinkedInOidc < OmniAuth::Strategies::OAuth2
+    option :name, "linkedin_oidc"
+
+    option :client_options,
+           {
+             site: "https://api.linkedin.com",
+             authorize_url: "https://www.linkedin.com/oauth/v2/authorization?response_type=code",
+             token_url: "https://www.linkedin.com/oauth/v2/accessToken",
+           }
+
+    option :scope, "openid profile email"
+    option :fields, %w[id full-name first-name last-name picture-url email-address]
+    option :redirect_url
+
+    uid { raw_info["sub"] }
+
+    info do
+      {
+        email: raw_info["email"],
+        first_name: raw_info["given_name"],
+        last_name: raw_info["family_name"],
+        picture_url: raw_info["picture"],
```

Comment:
Omniauth (and therefore Discourse) use 'image' as the key for this (per https://github.com/omniauth/omniauth/wiki/Auth-Hash-Schema)

```suggestion
        image: raw_info["picture"],
```